File: snakes-sabar/snakes-sabar.py
for _ in range(int(input())):
    n = int(input())
    n0 = 0
    n1 = 0
    b = input()
    for i in range(len(b)):
        if b[i] == '1':
            n1+=1
        else:
            n0+=1
    if n&1:
        if n0&1:
            print("Campus nahi banega")
            continue
        else:
            print(min(n0, n1))
            continue
    else:
        if n0&1:
            print(n1)
            continue
        else:
            print(n0)
            continue

File: snakes-sabar/snakes-sabar.c
#include <stdio.h>

int main(){
    long long int t;
    scanf("%lld", &t);
    while(t--) {
        long long int n, n0 = 0, n1 = 0;
        scanf("%lld", &n);
        char b[n+1];
        scanf("%s", b);
        for (int i = 0; i < n; i++){
            b[i] == '1'? n1++: n0++;
        }
        if (n&1) {
            if (n0&1) {
                printf("Campus nahi banega\n");
                continue;
            } else {
                long long int min = n0<n1?n0:n1;
                printf("%lld\n", min);
                continue;
            }
        } else {
            if (n0&1) {
                 printf("%lld\n", n1);
                 continue;
             } else {
                 printf("%lld\n", n0);
                 continue;
             }
        }
    }
    return 0;
}
